src/main/java/io/leonis/algieba/algebra/Ring.java

```diff
@@ -0,0 +1,60 @@
+package io.leonis.algieba.algebra;
+
+/**
+ * The Interface Ring.
+ *
+ * This interface describes the functionality of a <a href="https://en.wikipedia.org/wiki/Ring_(mathematics)">ring</a>.
+ * A ring is a mathematical object which has the following properties:
+ *
+ * - A ring is defined over a field (of elements), such as the real numbers or the complex numbers.
+ * - A ring has two operators: addition (+) and multiplication (*) - When an operator is applied to
+ * a number, it will result in a number from the same field (eg. addition of two reals results in a
+ * real, multiplication of two complex numbers yield a complex number, etc.)
+ *
+ * A ring is an Abelian {@link Group}, or a commutative group, with a second operator which is
+ * associative by itself, and distributes over the addition operator. A ring is governed by the
+ * following axioms (also known as the ring axioms):
+ *
+ * - Closure:        if a and b are elements of F, then a + b is an element of F - Associativity: if
+ * a, b, and c are elements of F, then (a+b)+c = a+(b+c) if a, b, and c are elements of F, then
+ * (a*b)*c = a*(b*c) - Identity:       there exists an element e of F such that for all a in F: e+a
+ * = a+e = a+0 - Inverse:        for every element, a, in F there exists another element, b, such
+ * that a*b=1 - Distributivity: if a, b, and c are elements of F, then a*(b+c) = a*b + a*c
+ *
+ * @param <F> The type of field element (eg. real, complex, integer, etc.)
+ * @author Rimon Oz
+ */
+public interface Ring<F> extends Group<F> {
+
+  /**
+   * @param <E>           the type parameter
```

Comment:
the type parameter isn't that descriptive; because I'm actually curious why you'd parameterize the exception-type.

Reply:
it's parametrized because java's way of dealing with division by zero is broken; ie. it doesn't throw an `ArithmeticException` (which is in the jdk) and there's no specific `DivisionByZeroException`. this way the implementation can specify it's own way of dealing with divisions which can't be computed. will update the javadoc though